server/src/assignment/assignment.service.ts

```diff
@@ -228,22 +215,35 @@ function computeStatusTab(project, lupteam) {
     return 'archive';
   }
 
-  if (participantProjectMilestones.find(milestone => milestone.statuscode === 'Not Started')) {
-    return 'upcoming';
-  }
-
   // TODO: this is sensitive to sort order. we need to revise this!!!
   if (
-    participantProjectMilestones.find(milestone => ['In Progress', 'Completed'].includes(milestone.statuscode))
-      && participantDispositions.find(disposition => ['Inactive'].includes(disposition.statecode) && ['Submitted', 'Not Submitted'].includes(disposition.statuscode))) {
+    participantDispositions.find(disposition =>
+      // ['General Public', 'LUP'].includes(disposition.dcp_visibility)
+      [717170003, 717170004].includes(disposition.dcp_visibility)
+      && ['Submitted', 'Not Submitted'].includes(disposition.statuscode)
+      && ['Inactive'].includes(disposition.statecode))
+    ) {
     return 'reviewed';
   }
 
   if (
-    participantProjectMilestones.find(milestone => ['In Progress', 'Completed'].includes(milestone.statuscode))
-      && participantDispositions.find(disposition => ['Active'].includes(disposition.statecode) && ['Draft', 'Saved'].includes(disposition.statuscode))) {
+    participantDispositions.find(disposition =>
+      // ['General Public', 'LUP'].includes(disposition.dcp_visibility)
+      [717170003, 717170004].includes(disposition.dcp_visibility)
+      && ['Draft', 'Saved'].includes(disposition.statuscode)
+      && ['Active'].includes(disposition.statecode))
+    ) {
     return 'to-review';
   }
 
+  if (
+    participantDispositions.find(disposition =>
+      disposition.dcp_visibility == null // upcoming if dcp_visibility is null
```

Comment:
nit: this comment can be removed, since the code is clear enough that it is self-documenting